TerminalApp/SSHConnection.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  SSHConnection.swift
+//  TerminalApp
+//
+//  Created by Yusuke Mori on 2022/12/12.
+//
+
+import Foundation
+import NMSSH
+
+// エラー理由を列挙
+enum SSHSessionError: Error {
+    case authorizationFailed
+    case connectFailed
+}
+
+class SSHConnection {
+    
+    var session: NMSSHSession
+    var password: String
+    
+    // 初期化処理
+    init(host: String, port: Int, andUsername: String, password: String) {
+        session = NMSSHSession(host: host, port: port, andUsername: andUsername)
+        self.password = password
+    }
+    
+    // インスタンス破棄時にセッションを閉じる
+    deinit {
+        disconnect()
+    }
+    
+    // SSH接続処理
+    func connect() throws {
+        if(session.connect()) { // SSHセッションを開始
```

Comment:
guardを使うと正常系処理をブロックで囲う必要がなくなり、すっきりするかもしれません。

```suggestion
       guard session.connect() else { throw  SSHSessionError.connectFailed } { // SSHセッションを開始
```